lib/liquid/context.rb

```diff
@@ -68,9 +67,22 @@ def handle_error(e, token = nil)
         e.set_line_number_from_token(token)
       end
 
+      output = nil
+
+      if exception_handler
+        result = exception_handler.call(e)
+        case result
+        when Exception
```

Comment:
I don't understand why this works. If you call `raise` above, doesn't that make the exception bubble up? Why does this work without a `rescue` here?